src/lib.rs

```diff
@@ -333,13 +351,18 @@ where
       .synthesize(&mut cs_primary)
       .map_err(|_| NovaError::SynthesisError)
       .expect("Nova error synthesis");
-    let (u_primary, w_primary) = cs_primary
-      .r1cs_instance_and_witness(&pp.circuit_shape_primary.r1cs_shape, &pp.ck_primary)
-      .map_err(|_e| NovaError::UnSat)
-      .expect("Nova error unsat");
+
+    let (u_primary, w_primary) = r1cs::instance_and_witness(
+      r1cs_primary,
+      &pp.ck_primary,
+      input_assignment,
+      aux_assignment,
+    )?;
```

Comment:
This new `r1cs::instance_and_witness` function eats the inputs instead of cloning.